service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java

```diff
@@ -0,0 +1,153 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import com.github.mustachejava.Mustache;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Attachment;
+import org.hl7.fhir.dstu3.model.BaseReference;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.DocumentReference;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.fhir.instance.model.api.IIdType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.NarrativeStatementTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DocumentReferenceToNarrativeStatementMapper {
+
+    private static final Mustache NARRATIVE_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_narrative_statement_template.mustache");
+    private static final String DEFAULT_ATTACHMENT_CONTENT_TYPE = "text/plain";
+    private final MessageContext messageContext;
+
+    public String mapDocumentReferenceToNarrativeStatement(final DocumentReference documentReference) {
+        final String narrativeStatementId = messageContext.getIdMapper()
+            .getOrNew(ResourceType.DocumentReference, documentReference.getId());
+
+        final NarrativeStatementTemplateParameters.NarrativeStatementTemplateParametersBuilder builder =
+            NarrativeStatementTemplateParameters.builder();
+
+        final Optional<Attachment> contentAttachment = documentReference.getContent().stream()
+            .map(DocumentReference.DocumentReferenceContentComponent::getAttachment)
+            .findFirst();
+
+        if (contentAttachment.isPresent() && contentAttachment.get().hasTitle()) {
+
+            final Attachment attachment = contentAttachment.get();
+            final String attachmentTitle = attachment.getTitle();
+            final String attachmentContentType = attachment.hasContentType()
+                ? attachment.getContentType() : DEFAULT_ATTACHMENT_CONTENT_TYPE;
+
+            builder.narrativeStatementId(narrativeStatementId)
+                .hasReference(true)
+                .comment(getComment(documentReference, attachmentTitle))
+                .availabilityTime(getAvailabilityTime(documentReference))
+                .referenceTitle(attachmentTitle)
+                .referenceContentType(attachmentContentType);
+        } else {
+            builder.narrativeStatementId(narrativeStatementId)
+                .comment(getComment(documentReference, null))
+                .availabilityTime(getAvailabilityTime(documentReference));
```

Comment:
This is the same on both conditional branches, so perhaps could be added to the builder creation part (lines 35,36).

Reply:
Good idea 👍